src/state/status/index.ts

```diff
@@ -1,82 +1,78 @@
 import { ViewService } from '@penumbra-zone/protobuf';
 import { penumbra } from '@/utils/penumbra';
 import { getSyncPercent } from '@/state/status/getSyncPercent';
-import { create } from 'zustand/index';
+import { makeAutoObservable, when } from 'mobx';
+import {
+  StatusResponse,
+  StatusStreamResponse,
+} from '@penumbra-zone/protobuf/penumbra/view/v1/view_pb';
+import { connectionStore } from '@/state/connection';
 
-export interface StatusState {
+class StatusState {
   /** If true, ignore all other state values */
-  loading: boolean;
+  loading = true;
   /** The error is set in case of a request failure */
   error?: string;
   /** Indicates that the account needs syncing with the blockchain */
-  syncing: boolean;
+  syncing = false;
   /** Indicates that the account is almost in sync with the blockchain (amount of unsynced blocks is less than 10) */
-  updating?: boolean;
+  updating?: boolean = false;
   /** The amount of synced blocks */
-  fullSyncHeight: bigint;
+  fullSyncHeight = 0n;
   /** The total amount of blocks in the blockchain */
   latestKnownBlockHeight?: bigint;
   /** A number between 0 and 1 indicating the sync progress */
-  syncPercent: number;
+  syncPercent = 0;
   /** A stringified sync percentage, e.g. '100%' or '17%' */
-  syncPercentStringified: string;
+  syncPercentStringified = '0%';
 
-  /**
-   * Receives the status stream from the view service
-   * and stores it in the `penumbraStatus` signal.
-   *
-   * Call this function in the global `useEffect` hook to subscribe to provider connection change
-   */
-  setup: () => Promise<void>;
-}
-
-/**
- * Initial status request doesn't return `latestKnownBlockHeight`.
- * Instead, it returns `catchingUp` to know if the sync is in progress
- */
-const getPenumbraStatus = async (): Promise<Omit<StatusState, 'setup'>> => {
-  const status = await penumbra.service(ViewService).status({});
+  constructor() {
+    makeAutoObservable(this);
 
-  return {
-    loading: false,
-    error: undefined,
-    syncing: status.catchingUp,
-    fullSyncHeight: status.fullSyncHeight,
-    latestKnownBlockHeight: status.catchingUp ? undefined : status.fullSyncHeight,
-    syncPercent: status.catchingUp ? 0 : 1,
-    syncPercentStringified: status.catchingUp ? '0%' : '100%',
-  };
-};
+    when(
+      () => connectionStore.connected,
+      () => void this.setup(),
+    );
+  }
```

Comment:
A substitute for https://github.com/penumbra-zone/dex-explorer/pull/77/files#diff-4c50d9dc437e2361505ebbcb744f036646280707449adeff19752b15f236c992L28-L33. Able to define store-to-store relationships/reactions outside of react.